src/source/RazorEngine.Core/Compilation/TypeContext.cs

```diff
@@ -56,6 +57,11 @@ internal TypeContext(string className, ISet<string> namespaces)
         /// Gets or sets the base template type.
         /// </summary>
         public Type TemplateType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the compiler references
+        /// </summary>
+        public ISet<CompilerReference> References { get; set; }
```

Comment:
On the one hand I think this should be internal. On the other hand the `ICompilerService` is an extension point so people should be able to do the same than the 'out of the box' implementation (in fact there is `RazorEngine.Roslyn`...). So there needs to be a point to add you own references. I think public is fine for now. (As long as it is undocumented it's kind of 'use it on your own risk').